hca/define_api.py

```diff
@@ -2,36 +2,133 @@
 
 import json, sys, os
 from io import open
+import pprint
 import requests
 from .parser import get_parser
 
 
 class API:
     """Class for interacting with the REST Api."""
 
-    def __init__(self, api_url, user):
+    def __init__(self, user, password, test=False):
         """Initialize the CLI API."""
-        self.spec = self.get_spec()
-        self.base_url = self.spec['host'] + self.spec['basePath']
-        self.parser = get_parser(self.spec)
+        spec = self.get_spec(test)
+        self.base_url = "https://" + spec['host'] + spec['basePath']
+        self.parser, self.param_data = get_parser(spec)
 
-    def get_spec(self):
+        self.positional_args = {endpoint: [arg['argument'] for arg in self.param_data[endpoint]['positional']] for endpoint in self.param_data}
+
+    def get_spec(self, test):
         """Load the API specification."""
-        with open(os.path.dirname(os.path.realpath(__file__)) + "/api_spec.json") as fp:
+        url = os.path.dirname(os.path.realpath(__file__)) + "/api_spec.json"
+        if test:
+            url = os.path.dirname(os.path.realpath(__file__)) + "/../test/test.json"
```

Comment:
Realized I'd never responded to these threads, but I've done each of the changes you recommended. In this case, api_spec is the one downloaded from the remote api whereas test.json is a spec that I've been using to test different features myself.